apps/asn1parse.c

```diff
@@ -178,7 +178,7 @@ int asn1parse_main(int argc, char **argv)
 
     if ((buf = BUF_MEM_new()) == NULL)
         goto end;
-    if (informat == FORMAT_PEM) {
+    if (genstr == NULL && informat == FORMAT_PEM) {
```

Comment:
Not just genstr but also the genconf option is broken in the same way.

Reply:
Yup, I'll fix that up